src/main/src/test/java/com/shevtsov/dao/impl/ClientDBDaoTest.java

```diff
@@ -0,0 +1,156 @@
+package com.shevtsov.dao.impl;
+
+import com.shevtsov.dao.ClientDao;
+import com.shevtsov.dao.DBConnection;
+import com.shevtsov.domain.Client;
+import org.junit.*;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class ClientDBDaoTest {
+    private DBConnection dbConnection = new DBConnectionInMemory();
+    private Connection connection;
+    private ClientDao clientDao;
+    private Client client;
+
+    @Before
+    public void setUp() {
+        clientDao = new ClientDBDao(dbConnection);
+        client = new Client(1, "name", "surname", 0, "phone", "email");
+        try {
+            connection = dbConnection.getConnection();
+            try (PreparedStatement statement = connection.prepareStatement(
+                    "CREATE TABLE IF NOT EXISTS CLIENTS (ID BIGINT PRIMARY KEY AUTO_INCREMENT," +
```

Comment:
Yes. But this content is lost after the constructor completion. To avoid this, I can use the URL postfix "; DB_CLOSE_DELAY = -1". But then I will have one content for all tests, and the tests will affect each other. Now I create new tables before each test, and after the test all data is cleared when the connection is closed. Thus, each test is independent. Is this way acceptable?